### PR TITLE
Expose a debug host for local lambda

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
Values are queried from Docker settings in IDE

## Motivation and Context
Allows connecting to a docker host which is using VirtualBox or any other IP

## Testing
Exposed Docker for Mac over tcp socket
```
docker run -d -v /var/run/docker.sock:/var/run/docker.sock -p 127.0.0.1:1234:1234 bobrik/socat TCP-LISTEN:1234,fork UNIX-CONNECT:/var/run/docker.sock
```

```
{"detail":null,"detail-type":"HelloWorldFunction updated event of null","resources":null,"id":null,"source":null,"time":null,"region":null,"version":null,"account":null}
END RequestId: 75f41372-cbbf-16d1-2f46-cdfa35eeff6c
REPORT RequestId: 75f41372-cbbf-16d1-2f46-cdfa35eeff6c	Init Duration: 1039.17 ms	Duration: 2252.94 ms	Billed Duration: 2300 ms	Memory Size: 128 MB	Max Memory Used: 31 MB	
Disconnected from the target VM, address: '127.0.0.9:49976', transport: 'socket'
```

## Screenshots (if appropriate)
Screen Shot 2020-01-03 at 2 01 11 PM

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamDebugSupport.kt [hunk 1]
```diff
@@ -141,11 +144,13 @@ class DotNetSamDebugSupport : SamDebugSupport {
...
         val startInfo = createNetCoreStartInfo(state)
 
+        val debugHostAddress = InetAddress.getByName(debugHost)
+
         val protocol = Protocol(
             serializers = Serializers(),
             identity = Identities(IdKind.Client),
             scheduler = scheduler,
-            wire = SocketWire.Client(debuggerLifetime, scheduler, port = frontendPort, optId = "FrontendToDebugWorker"),
+            wire = SocketWire.Client(debuggerLifetime, scheduler, hostAddress = debugHostAddress, port = frontendPort, optId = "FrontendToDebugWorker"),
```
Comment (abrooksv) [inline jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamDebugSupport.kt]: @sdubov Do I need to pass host anywhere for the backend port?

**Comment by kiiadi** [comment]
Assume this is a replacement for #1402?

**Comment by abrooksv** [comment]
> Assume this is a replacement for #1402?

Correct, just waiting for Sergey to confirm the Rider changes

File: jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamDebugSupport.kt (hunk 1 above)
Comment (sdubov) [reply jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamDebugSupport.kt]: We start debugger worker in Server mode. It starts listening for any host to connect when we start. Should work as is.

**Pull request merged**